src/main/java/com/hobbycollection/HobbyCollectionRESTController.java

```diff
@@ -2,34 +2,47 @@
 
 import com.hobbycollection.dto.Collection;
 import com.hobbycollection.service.ICollectionService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.*;
 
 @RestController
 public class HobbyCollectionRESTController {
     @Autowired
     private ICollectionService collectionService;
+    Logger log = LoggerFactory.getLogger(this.getClass());
 
     /**
      * Fetches a collection by the id
      * @param id
      * @return URL
      */
     @GetMapping("/api/Collection/fetchByid")
-    public Collection collectionFetchById(@RequestParam(value="id") int id){
-        return new Collection("test name", "http://some.url","test description");
+    public Object collectionFetchById(@RequestParam(value="id") int id){
+        try{
+            log.info("Collection with ID of " + id + " has been fetched.");
+            return new Collection("test name", "http://some.url","test description");
+        }
+        catch (Exception e){
+            log.error("There was en error finding your collection with ID: " + id);
```

Comment:
In this specific case, it's unlikely that an exception would be thrown from an object instantiation, but this is a good pattern to follow, none the less.  Good segregation to error log.

Also, include the exception in the log method call.